### PR TITLE
Add sendinblue integration

### PR DESCRIPTION
**Pull request by luca-rath**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #issuenum
| Related issues/PRs | #issuenum
| License | MIT

#### What's in this PR?

Add sendinblue integration

#### Example Usage

See `sendinblue.md` file

#### To Do

- [x] Create documentation

### CODE REVIEW

File: Resources/config/form-fields/field_sendinblue.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <property name="options/attributeNameFirstName" type="text_line" colspan="6">
+        <meta>
+            <title>sulu_form.sendinblue_attribute_name_first_name</title>
+        </meta>
+    </property>
+    <property name="options/attributeNameLastName" type="text_line" colspan="6">
+        <meta>
+            <title>sulu_form.sendinblue_attribute_name_last_name</title>
+        </meta>
+    </property>
```
Comment (alexander-schranz) [inline Resources/config/form-fields/field_sendinblue.xml]: This should be as simple as possible and we should here just go in the listener with FIRST_NAME and LAST_NAME defaults like we do it in mailchimp also.

File: Resources/config/form-fields/field_sendinblue.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                    type="expression"
+                    value="service('sulu_form.dynamic.sendinblue_mail_template_select').getValues()"
+            />
+        </params>
+    </property>
+    <property name="options/redirectionUrl" type="url" mandatory="true">
+        <meta>
+            <title>sulu_form.sendinblue_redirection_url</title>
+        </meta>
+    </property>
```
Comment (alexander-schranz) [inline Resources/config/form-fields/field_sendinblue.xml]: What is this for? Do we need it? Can we just not use the current page url or even no url. Should not sending blue UI define this.

Comment (luca-rath) [reply Resources/config/form-fields/field_sendinblue.xml]: This is required by the sendinblue api ... But yes, I could just use the current url

**Comment by luca-rath** [comment]
@alexander-schranz fixed everything

File: Dynamic/Helper/SendinblueListSelect.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\FormBundle\Dynamic\Helper;
+
+use SendinBlue\Client\Api\ContactsApi;
+use SendinBlue\Client\Configuration;
+
+class SendinblueListSelect
```
Comment (alexander-schranz) [inline Dynamic/Helper/SendinblueListSelect.php]:
```suggestion

/**
 * @final
 * @internal
 */
class SendinblueListSelect
```

File: Dynamic/Helper/SendinblueMailTemplateSelect.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\FormBundle\Dynamic\Helper;
+
+use SendinBlue\Client\Api\TransactionalEmailsApi;
+use SendinBlue\Client\Configuration;
+
+class SendinblueMailTemplateSelect
```
Comment (alexander-schranz) [inline Dynamic/Helper/SendinblueMailTemplateSelect.php]:
```suggestion

/**
 * @final
 * @internal
 */
class SendinblueMailTemplateSelect
```

File: Event/SendinblueListSubscriber.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+namespace Sulu\Bundle\FormBundle\Event;
+
+use SendinBlue\Client\Api\ContactsApi;
+use SendinBlue\Client\Configuration;
+use SendinBlue\Client\Model\CreateDoiContact;
+use Sulu\Bundle\FormBundle\Entity\Dynamic;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class SendinblueListSubscriber implements EventSubscriberInterface
```
Comment (alexander-schranz) [inline Event/SendinblueListSubscriber.php]:
```suggestion

/**
 * @final
 * @internal
 */
class SendinblueListSubscriber implements EventSubscriberInterface
```

File: Event/SendinblueListSubscriber.php [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        if (!$request) {
+            return;
+        }
+
+        $form = $dynamic->getForm()->serializeForLocale($dynamic->getLocale(), $dynamic);
+
+        $email = '';
+        $firstName = '';
+        $lastName = '';
+        $redirectionUrl = $request->getUriForPath('');
```
Comment (alexander-schranz) [inline Event/SendinblueListSubscriber.php]: @luca-rath should we add`?subscribe=true`. So in the frontend a message can be shown?

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
Some small change requests.

File: Event/SendinblueListSubscriber.php (hunk 6 above)
Comment (luca-rath) [reply Event/SendinblueListSubscriber.php]:
```suggestion
        $redirectionUrl = $request->getUriForPath('') . '?subscribe=true';
```

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**